corehq/apps/data_dictionary/models.py

```diff
@@ -98,6 +100,10 @@ def save(self, *args, **kwargs):
         get_data_dict_props_by_case_type.clear(self.case_type.domain)
         return super(CaseProperty, self).save(*args, **kwargs)
 
+    def clean(self):
+        if self.required and self.deprecated:
+            raise ValidationError(_("A required property can't be deprecated"))
```

Comment:
I'm ambivalent about this. Why make the user unrequire the property before deprecating it?

Reply:
I was in split about this as well.

The reason I considered this was because when we do case import and select fields to validate
https://github.com/dimagi/commcare-hq/blob/baed5e6f077231cf47fd21ba444246fd2ff21659/corehq/apps/case_importer/do_import.py#L91, we don't filter out deprecated properties https://github.com/dimagi/commcare-hq/blob/50bcae5ac384900f982c69af8dab550862ea839e/corehq/apps/data_dictionary/util.py#L230
So this would lead to fields getting validated as required. Hence this check here.

Reply:
I was not sure of updating fields_to_validate to filter out deprecated properties.